aws/cloudwatch_dashboard.go

```diff
@@ -0,0 +1,89 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/gruntwork-io/go-commons/errors"
+
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+)
+
+func getAllCloudWatchDashboards(session *session.Session, excludeAfter time.Time, configObj config.Config) ([]*string, error) {
+	svc := cloudwatch.New(session)
+
+	allDashboards := []*string{}
+	input := &cloudwatch.ListDashboardsInput{}
+	err := svc.ListDashboardsPages(
+		input,
+		func(page *cloudwatch.ListDashboardsOutput, lastPage bool) bool {
+			for _, dashboard := range page.DashboardEntries {
+				if shouldIncludeCloudWatchDashboard(dashboard, excludeAfter, configObj) {
+					allDashboards = append(allDashboards, dashboard.DashboardName)
+				}
+			}
+			return !lastPage
+		},
+	)
+	return allDashboards, errors.WithStackTrace(err)
+}
+
+func shouldIncludeCloudWatchDashboard(dashboard *cloudwatch.DashboardEntry, excludeAfter time.Time, configObj config.Config) bool {
+	if dashboard == nil {
+		return false
+	}
+
+	if dashboard.LastModified != nil && excludeAfter.Before(*dashboard.LastModified) {
+		return false
+	}
+
+	return config.ShouldInclude(
+		aws.StringValue(dashboard.DashboardName),
+		configObj.CloudWatchDashboard.IncludeRule.NamesRegExp,
+		configObj.CloudWatchDashboard.ExcludeRule.NamesRegExp,
+	)
+}
+
+func nukeAllCloudWatchDashboards(session *session.Session, identifiers []*string) error {
+	region := aws.StringValue(session.Config.Region)
+
+	svc := cloudwatch.New(session)
+
+	if len(identifiers) == 0 {
+		logging.Logger.Infof("No CloudWatch Dashboards to nuke in region %s", region)
+		return nil
+	}
+
+	// NOTE: we don't need to do pagination here, because the pagination is handled by the caller to this function,
```

Comment:
just to confirm - is that a pattern we should adopt more or less in all cases for pagination?

Reply:
Yes I think we should. I've been doing this on any new resource I introduce, but haven't been going back and retroactively porting it.

Reply:
Hm, that's awesome! I've raised an issue actually some weeks ago about this: https://github.com/gruntwork-io/cloud-nuke/issues/226 so really good to see this!